packages/firestore/src/model/mutation.ts

```diff
@@ -42,7 +42,7 @@ import { TransformOperation } from './transform_operation';
  *             containing foo
  */
 export class FieldMask {
-  private constructor(readonly fields: SortedSet<FieldPath>) {
+  constructor(readonly fields: SortedSet<FieldPath>) {
```

Comment:
For consistency with fromArray (as well as for consistency with the other platforms) you might consider leaving the ctor private and adding a 'fromSet' method.

In particular, the android sdk has (copy+pasted):

```
public final class FieldMask {
  public static FieldMask fromSet(Set<FieldPath> mask) {
    return new FieldMask(mask);
  }

  private final Set<FieldPath> mask;

  private FieldMask(Set<FieldPath> mask) {
    this.mask = mask;
  }
```

OTOH, the C++ just uses the ctor, so you'll end up being inconsistent with something, regardless of which you choose. :)

Optional.

Reply:
Done.